Personal project/src/ro/scit/evpp/Dealership.java

```diff
@@ -0,0 +1,134 @@
+package ro.scit.evpp;
+
+import java.util.*;
+
+public class Dealership {
+
+    private HashSet<Car> cars = new HashSet<>();
+    private HashMap<Car, Integer> soldCars = new HashMap<>();
+    private HashMap<Car, Integer> stockOfCars = new HashMap<>();
+    private HashMap<Customer, Car> historyOfSales = new HashMap<>();
+    private int request = 0;
+    private float revenue;
+
+
+    public void addCar(Car c, int stockNumber) {
+        if (cars.contains(c)) {
+            increaseStockNumber(c, stockNumber);
+        } else {
+            addNewCar(c, stockNumber);
+        }
+    }
+
+    public void addNewCar(Car c, int stockNumber) {
+        cars.add(c);
+        stockOfCars.put(c, stockNumber);
+    }
+
+    private void increaseStockNumber(Car c, int stockNumber) {
+        int count = stockOfCars.get(c) + stockNumber;
+        stockOfCars.put(c, count);
+    }
+
+    public void removeCar(Car c, int stockNumber) {
+        int count = stockOfCars.get(c);
+        if (count == 1) {
+            removeNewCar(c, stockNumber);
+        } else {
+            decreaseStockNumber(c, stockNumber);
+        }
+    }
+
+    public void removeNewCar(Car c, int stockNumber) {
```

Comment:
shouldn't this method be private? will it be used by a class who uses the Dealership class?